domain-ee/ee-ep-merge-app/src/python_src/util/metric_logger.py

```diff
@@ -31,66 +35,98 @@
 distribution_metrics_api = metrics_v1.MetricsApi(api_client)  # Metrics API does not have an endpoint for distribution metrics
 
 
-def increment(metric: str, value: float = 1) -> None:
+class Metric:
     """
-    Increments a count metric with by the name 'APP_PREFIX.{metric}'
-    :param metric: string containing the metric name
-    :param value: value to increment by
+    A class used to represent a base Metric.
+
+    Attributes:
+        name (str): The name of the metric
+        value (float): The value of the metric
     """
-    full_metric = f'{APP_PREFIX}.{metric.strip(".").lower()}'
-
-    body = MetricPayload(
-        series=[
-            MetricSeries(
-                metric=full_metric,
-                type=MetricIntakeType.COUNT,
-                points=[
-                    MetricPoint(
-                        timestamp=int(datetime.now().timestamp()),
-                        value=value,
-                    ),
-                ],
-                tags=STANDARD_TAGS,
-            )
-        ],
-    )
+
+    def __init__(self, name: str, value: float):
+        self.name = name
+        self.value = value
+
+    def __eq__(self, other):
+        if not isinstance(other, self.__class__):
+            return False
+        return self.name == other.name and self.value == other.value
+
+    def __hash__(self):
+        return hash((self.name, self.value))
```

Comment:
It doesn't look like you need any nuanced mutability stuff with name and value, so you could use Python's included battery [typing.NamedTuple](https://docs.python.org/3/library/typing.html#typing.NamedTuple) for this.